### ISSUE TITLE
instance destroy does a INSERT INTO instead of an UPDATE

### ISSUE THREAD

**Issue by JSteunou** [sequelize/sequelize#2907]
I have this (weird) test that fail with #master 

```javascript
            db.User.bulkCreate(users).then(function(dbUsers) {
                // delete odd users
                return db.Sequelize.Promise.all(
                    dbUsers.filter(function(user) {
                        return user.id % 2 === 1;
                    }).map(function(user) {
                        return user.destroy();
                    })
                );
            }).then(function() {
                 ....
            }).catch(function(e) {
                console.log('======= ERROR', e);
                throw e;
            });
```

Because the destroy does an insert into instead of an update.

```
Executing (default): INSERT INTO `user` (`id`,`created_at`,`updated_at`,`deleted_at`,`email`) VALUES (3,'2015-01-12 10:24:56','2015-01-12 10:24:56','2015-01-12 10:24:56','jean.dupont@gmail.com');
======= ERROR { [SequelizeUniqueConstraintError: Validation error]
  name: 'SequelizeUniqueConstraintError',
  message: 'Validation error',
  errors: 
   [ { message: 'PRIMARY must be unique',
       type: 'unique violation',
       path: 'PRIMARY',
       value: '1' } ],
  fields: { PRIMARY: '1' } }
```

**Comment by mickhansen** [comment]
Unless you are using postgres `bulkCreate` won't return instances with primary keys, this is a limitation of the respective databases.

**Comment by mickhansen** [comment]
If you are in fact using postgres you have to remember to use the `returning: true` option: `bulkCreate([rows], {returning: true})`

**Comment by mickhansen** [comment]
(deleted previous comments)

It appears that bulkCreate does not accurately set the instances as `isNewRecord: false` on bulkCreate even when the primary key is correctly returned (on pg with `returning: true`)

**Comment by kytwb** [comment]
Encountered this as well :+1:

**Comment by mickhansen** [comment]
@kytwb loop over the instances returned from bulkCreate and force them to have `isNewRecord: false`

**Comment by kytwb** [comment]
@mickhansen thanks.

**Comment by JSteunou** [comment]
Any news on this one?

**Comment by JSteunou** [comment]
Closed by b160a0c24f for my test case
